src/test/txpackage_tests.cpp

```diff
@@ -557,6 +577,182 @@ BOOST_FIXTURE_TEST_CASE(package_witness_swap_tests, TestChain100Setup)
+        const auto submit_package_too_low = ProcessNewPackage(m_node.chainman->ActiveChainstate(), *m_node.mempool,
+                                                   package_still_too_low, /* test_accept */ false);
+        BOOST_CHECK_MESSAGE(submit_package_too_low.m_state.IsInvalid(), "Package validation unexpectedly succeeded");
+        BOOST_CHECK_EQUAL(submit_package_too_low.m_state.GetResult(), PackageValidationResult::PCKG_POLICY);
+        BOOST_CHECK_EQUAL(submit_package_too_low.m_state.GetRejectReason(), "package-fee-too-low");
+        BOOST_CHECK_EQUAL(m_node.mempool->size(), expected_pool_size);
+        const CFeeRate child_feerate(200, GetVirtualTransactionSize(*tx_child_cheap));
+        BOOST_CHECK(child_feerate.GetFeePerK() > 1000);
+        const CFeeRate expected_feerate(200,
+            GetVirtualTransactionSize(*tx_parent_cheap) + GetVirtualTransactionSize(*tx_child_cheap));
+        BOOST_CHECK(expected_feerate.GetFeePerK() < 1000);
+        BOOST_CHECK(submit_package_too_low.m_package_feerate != std::nullopt);
+        BOOST_CHECK_MESSAGE(submit_package_too_low.m_package_feerate.value() == expected_feerate,
+                            strprintf("Expected package feerate %s, got %s", expected_feerate.ToString(),
+                                      submit_package_too_low.m_package_feerate.value().ToString()));
+    }
+
+    // Package feerate includes the modified fees of the transactions.
+    // This means a child with its fee delta from prioritisetransaction can pay for a parent.
+    m_node.mempool->PrioritiseTransaction(tx_child_cheap->GetHash(), 1 * COIN);
+    // Now that the child's fees have "increased" by 1 BTC, the cheap package should succeed.
+    {
+        const auto submit_prioritised_package = ProcessNewPackage(m_node.chainman->ActiveChainstate(), *m_node.mempool,
+                                                                  package_still_too_low, /*test_accept=*/ false);
+        expected_pool_size += 2;
+        BOOST_CHECK_MESSAGE(submit_prioritised_package.m_state.IsValid(),
+                "Package validation unexpectedly failed" << submit_prioritised_package.m_state.GetRejectReason());
+        const CFeeRate expected_feerate(1 * COIN + 200,
+            GetVirtualTransactionSize(*tx_parent_cheap) + GetVirtualTransactionSize(*tx_child_cheap));
+        BOOST_CHECK(submit_prioritised_package.m_package_feerate != std::nullopt);
+        BOOST_CHECK_MESSAGE(submit_prioritised_package.m_package_feerate.value() == expected_feerate,
+                            strprintf("Expected package feerate %s, got %s", expected_feerate.ToString(),
+                                      submit_prioritised_package.m_package_feerate.value().ToString()));
+    }
+
+    // Package feerate is calculated without topology in mind; it's just aggregating fees and sizes.
+    // However, this should not allow parents to pay for children. Each transaction should be
```

Comment:
"However, this should not allow parents to pay for children"

This reads to me like "parents must not pay for children". When it is more (to my understanding), parents must be validated first before considering children. If the parent(s) is accepted into the mempool first a low fee child can later join them as part of a sufficient fee package but a low fee child must not prevent a sufficient fee parent from being accepted into the mempool. That's right yeah?

Reply:
Or maybe because in the process of a child (subsidized by a parent) being accepted into the mempool the parent could be mined on its own and then the child would need to be kicked out the mempool?

Reply:
We shouldn't accept the child into the mempool if it doesn't meet the minimum feerate.

Reply:
Ah its a miner's economic interest argument. Ok gotcha, makes sense. Thanks